plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorSmokeTest.java

```diff
@@ -107,4 +112,63 @@ public void testS3LocationWithTrailingSlash()
 
         assertUpdate("DROP TABLE " + tableName);
     }
+
+    @Test
+    public void testExpireSnapshots()
+    {
+        String tableName = "test_expiring_snapshots_" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = prepareCleanUpSession();
+        String location = "s3://%s/%s/%s/".formatted(bucketName, schemaName, tableName);
+        List<Event> events = new ArrayList<>();
+        hiveMinioDataLake.getMinioClient().captureBucketNotifications(bucketName, event -> {
+            if (event.eventType().toString().toLowerCase(ENGLISH).contains("remove")) {
+                events.add(event);
+            }
+        });
+
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer) WITH (location='" + location + "')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 2)", 1);
+        assertThat(query("SELECT sum(value), listagg(key, ' ') WITHIN GROUP (ORDER BY key) FROM " + tableName))
+                .matches("VALUES (BIGINT '3', VARCHAR 'one two')");
+
+        List<String> initialMetadataFiles = hiveMinioDataLake.getMinioClient().listObjects(bucketName, "/%s/%s/metadata".formatted(schemaName, tableName));
+        assertThat(initialMetadataFiles).isNotEmpty();
+
+        List<Long> initialSnapshots = getSnapshotIds(tableName);
+        assertThat(initialSnapshots).isNotEmpty();
+
+        assertQuerySucceeds(sessionWithShortRetentionUnlocked, "ALTER TABLE " + tableName + " EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')");
+
+        List<String> updatedMetadataFiles = hiveMinioDataLake.getMinioClient().listObjects(bucketName, "/%s/%s/metadata".formatted(schemaName, tableName));
+        assertThat(updatedMetadataFiles).isNotEmpty().hasSizeLessThan(initialMetadataFiles.size());
+
+        List<Long> updatedSnapshots = getSnapshotIds(tableName);
+        assertThat(updatedSnapshots.size()).isLessThan(initialSnapshots.size());
```

Comment:
remove this since on next line you check it's precisely 1.
 instead, assert (earlier) that  `initialSnapshots.size()` is more than 1

btw use `assertThat(collection).hasSize`, so that in case of failure, the exception message carries the collection contents.